sms_toolkit/utils/encoding.py

```diff
@@ -26,10 +30,20 @@ def convert_to_unicode_characters(string):
 
 
 def encode_unicode_character_to_utf16(character):
-    if len(character) > 1:
-        return flatten(map(encode_unicode_character_to_utf16, character))
-
-    return [((0xff00 & ord(character)) >> 8), 0x00ff & ord(character)]
+    """
+    Encode the given unicode code points into utf-16 and return a list of bytes. Note that for code-points within
+    the BMP the number of bytes would be 2, but for code-points not in BMP, the number of bytes will be 4.
+
+    Args:
+        character: A unicode code-point
+
+    Returns:
+        A list of the bytes
+    """
+    if is_python_3():
+        return [c for c in character.encode('utf-16-be')]
```

Comment:
You don't need ord here?

Reply:
Nope. Encode returns bytes in py3. From the comment above - 

> str.encode behaves differently in python 2 and python 3.
> In python 2 encode returns an 8-bit string version of the Unicode string, encoded in the requested encoding (<a href="https://docs.python.org/2/howto/unicode.html">Source</a>).
> In python 3, encode returns an encoded version of the string as a bytes object. (<a href="https://docs.python.org/3/library/stdtypes.html#str.encode">Source</a>)
> 
> We basically want the list of byte values for the given character here. In python 3, we directly get it by iterating through every entry of encode. In py2, since the value is a string version, we can get the value by <code>ord</code>ing it.